app/models/preference_code.rb

```diff
@@ -5,4 +5,67 @@ def initialize(id:, description:)
   end
 
   attr_accessor :id, :description
+
+  def self.find(commodity, measure)
```

Comment:
Wow, this looks really complicated :) 
we should aim for a simpler implementation, which I don't know yet 😄

Reply:
maybe could we split logically this into different methods,
I noticed that some internal conditions (IF) is based on the `commodity` other on the `measure`, and some has not conditions, this seems interesting 🤔